shuup/admin/modules/media/views.py

```diff
@@ -141,37 +161,65 @@ def post(self, request, *args, **kwargs):
 
     def handle_get_folders(self, data):
         shop = get_shop(self.request)
-        root_folders = get_cached_trees(Folder._tree_manager.filter(_get_folder_query_filter(shop)))
-        return JsonResponse({"rootFolder": filer_folder_to_json_dict(None, root_folders)})
+        users_root_folder = Folder.objects.filter(media_folder__root_folder_for=self.request.user).first()
+        if users_root_folder:
+            root_folders = get_cached_trees(
+                Folder._tree_manager.filter(_get_folder_query_filter(shop, self.user) & ~Q(id=users_root_folder.id))
+            )
+            return JsonResponse({"rootFolder": filer_folder_to_json_dict(users_root_folder, root_folders)})
+        else:
+            root_folders = get_cached_trees(Folder._tree_manager.filter(_get_folder_query_filter(shop, self.user)))
+            return JsonResponse({"rootFolder": filer_folder_to_json_dict(None, root_folders)})
+
+    def handle_get_path(self, data):
+        shop = get_shop(self.request)
+        current_folder = _get_folder_query(shop, self.user).get(pk=data["id"])
+        path = []
+        for folder in current_folder.logical_path:
+            path.append(folder.name)
+
+        path.append(current_folder.name + "/")
+        return JsonResponse({"folderPath": "/".join(path)})
 
     def handle_post_new_folder(self, data):
         shop = get_shop(self.request)
         parent_id = int(data.get("parent", 0))
         if parent_id > 0:
-            parent = _get_folder_query(shop).get(pk=parent_id)
+            parent = _get_folder_query(shop, self.user).get(pk=parent_id)
         else:
             parent = None
 
-        name = data["name"]
+        if parent and subfolder_of_users_root(self.user, parent):
+            return self._create_folder(data["name"], parent, shop)
+        elif has_permission(self.user, "media.create-folder"):
```

Comment:
Doesn't this mean that I can always add folders anywhere if I have "create-folder" perissions.

Reply:
Yes

Reply:
But if you don't have that permission you will only be able to create folders that are in the sub folder tree of your root folder